validation-test/stdlib/FloatingPointConversion.swift.gyb

```diff
@@ -0,0 +1,159 @@
+
+}%
+
+func getInfiniteOrNaNMessage(_ floatType: String) -> String {
+  if _isDebugAssertConfiguration() {
+    return "either infinite or NaN"
+  }
+  return ""
+}
+
+func getTooSmallMessage() -> String {
+  if _isDebugAssertConfiguration() {
+    return "would be less than"
+  }
+  return ""
+}
+
+func getTooLargeMessage() -> String {
+  if _isDebugAssertConfiguration() {
+    return "would be greater than"
+  }
+  return ""
+}
+
+var FixedPointConversionTraps = TestSuite("FloatingPointToFixedPointConversionTraps")
+var FixedPointConversionFailures = TestSuite("FloatingPointToFixedPointConversionFailures")
+
+var FloatingPointConversionTruncations = TestSuite("FloatingPointToFloatingPointConversionTruncations")
+var FloatingPointConversionFailures = TestSuite("FloatingPointToFloatingPointConversionFailures")
+
+% for Self, selfSignificandBits in floatNameToSignificandBits.iteritems():
+
+%     if Self == 'Float80':
+#if arch(i386) || arch(x86_64)
+%     end
+
+%   for OtherFloat, otherSignificandBits in floatNameToSignificandBits.iteritems():
+
+%     if OtherFloat == 'Float80':
+#if arch(i386) || arch(x86_64)
+%     end
+
+%     if otherSignificandBits <= selfSignificandBits:
+
+/// Always-safe conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that nothing interesting happens and we end up with the same result after converting.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  let result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectEqual(input, resultConvertedBack)
+}
+
+/// Never-nil failable conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that nothing interesting happens and we end up with a non-nil, identical result.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(exactly: input)
+  expectNotEqual(result, nil)
+}
+
+%    else:
+
+/// Always-truncating conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we fail and crash when an float would be truncated in conversion.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+/// Always-nil failable conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we return nil when a float would be truncated in conversion.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
```

Comment:
`greatestFiniteMagnitude` would be out of range.  Could you check one that is in range, but would be truncated?`(1.0 as OtherFloat).nextUp`, for example.